lib/zendesk_apps_support/manifest/location_options.rb

```diff
@@ -24,6 +24,7 @@ def initialize(location, options)
         RUBY_TO_JSON.each do |ruby, json|
           instance_variable_set(:"@#{ruby}", options[json])
         end
+        @url = @url.to_s
```

Comment:
Rather than coercing to a string should we be throwing an ArgumentError if the url isn't a string?

Reply:
If we throw an ArgumentError here, we'd have to catch it and somehow bubble up a validation message (despite this file not being validation-related). IMO it's better to coerce to string and validate elsewhere that the urls look like urls. If I were an app developer I'd appreciate a `"true" is not a valid URL` error message more than a `parameter type invalid` message.

Reply:
Does the same issue occur if a boolean is passed in other places in the manifest? If so maybe we should add validations that the values are set to the correct types

Reply:
The problem here is due to this interesting logic in `location_options.rb`. I don't think it's an issue throughout the manifest.

Reply:
I've had a look at the code and the original airbrake. It looks like the `Manifest` class is used to hold the user input and access it during the validation step. Given that we're pre-validation it seems reasonable for the variable to be set to whatever the user entered, so you're right that we shouldn't be throwing an `ArgumentError`.

Instead I think perhaps we should be doing basic validations on the shape and types of values in the manifest before we do more complex validations that make assumptions about those types. So I think a fix that addresses the root cause is probably adding something like this to `ZendeskAppsSupport::Validations::Manifest#collate_manifest_errors`:

``` ruby
if has_valid_locations?(manifest)
 errors << ban_no_template(manifest) if manifest.iframe_only?
else
  errors << invalid_location_error(manifest)
end
```

Then `ban_no_template`, which expects the location to be a string, is only called if the location is a string. If the user has entered the wrong type then `invalid_location_error` will give them a helpful and specific error message.

Reply:
``` ruby
            if location_options.url.is_a?(String) && !location_options.url.empty?
```